DP-39184 Update glossary regex matching

andy-blum opened the pull request:
**Description:**
Updates the regex matching for glossary terms. Regexes now replace any punctuation in search terms with `.?`, allowing them to match any (or no) character.

For "clerk’s office", this would become `clerk.?s office`. This pattern will match:
* `clerk's office`
* `clerk’s office` (note the different apostrophe character
* `clerk s office`
* `clerk9s office`
* `clerks office` (with nothing in place of the apostrophe).


**Jira:** (Skip unless you are MA staff)
DP-39184


**To Test:**
- [ ] Add steps to test this feature


---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

andy-blum committed:
update glossary regex matching

andy-blum committed:
DP-39184: remove whitespace within popover template

andy-blum committed:
DP-39184: Clone popover element instead of all template content

andy-blum committed:
DP-39184: update mayflower deps

andy-blum committed:
DP-39184: add changelog

andy-blum committed:
Merge branch 'develop' into fix/glossary-matching

andy-blum committed:
unrelated phpcs error

andy-blum committed:
Add test for punctuation injecting extra space

andy-blum committed:
revert test changes

andy-blum committed:
fix injection test

dstorozhuk reviewed (approved):
LGTM, waiting for MF PR to be merged.

dstorozhuk committed:
Update dependency version for mayflower-artifacts

Switched massgov/mayflower-artifacts from "dev-fix/DP-39184_fix-glossary-term-matching" to "dev-develop" in

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>
